cirq-ft/cirq_ft/algos/arithmetic_gates.py

```diff
@@ -130,6 +131,135 @@ def _t_complexity_(self) -> infra.TComplexity:
         )
 
 
+@attr.frozen
+class BiQubitsMixer(infra.GateWithRegisters):
+    """Implements the COMPARE2 (Fig. 1) https://www.nature.com/articles/s41534-018-0071-5#Sec8
+
+    This gates mixes the values in away that preserves the result of comparison.
+    The registers being compared are 2-qubit registers where
+        x = 2*x_msb + x_lsb
+        y = 2*y_msb + y_lsb
+    The Gate mixes the 4 qubits so that sign(x - y) = sign(x_lsb' - y_lsb') where x_lsb' and y_lsb'
+    are the final values of x_lsb' and y_lsb'.
+    """
+
+    adjoint: bool = False
+
+    @cached_property
+    def registers(self) -> infra.Registers:
+        return infra.Registers.build(x=2, y=2, ancilla=3)
+
+    def _has_unitary_(self):
+        return not self.adjoint
+
+    def decompose_from_registers(
+        self, *, context: cirq.DecompositionContext, **quregs: Sequence[cirq.Qid]
+    ) -> cirq.OP_TREE:
+
+        x, y, ancilla = quregs['x'], quregs['y'], quregs['ancilla']
```

Comment:
I'm pretty sure the style is (maybe it changed?) to replace `**quregs` in the decompose function signature with `x, y, ancilla` which will implicitly do the dictionary unpacking